### PR TITLE
feat: default tx subscription resolution config

### PR DESCRIPTION
**Pull request by rflechtner**
## fixes KILTProtocol/ticket#2185

Adding a config service option to set a default for the resolveOn parameter on `submitTx`, which will be used whenever this parameter is not set.
Also inlining the `parseSubscriptionOption` function which had little to no purpose at this point

## How to test:

Tests

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

**Review by tjwelde — APPROVED** [review]
LGTM

File: packages/chain-helpers/src/blockchain/Blockchain.ts [hunk 1]
```diff
@@ -88,53 +88,45 @@ export function EXTRINSIC_FAILED(result: ISubmittableResult): boolean {
...
-      EXTRINSIC_FAILED(result) || IS_ERROR(result),
-    timeout,
-  } = { ...opts }
-
-  return {
-    resolveOn,
-    rejectOn,
-    timeout,
-  }
+function defaultResolveOn(): SubscriptionPromise.ResultEvaluator {
```
Comment (ntn-x2) [inline packages/chain-helpers/src/blockchain/Blockchain.ts:91]: Could this be a getter? So we could access it like `Blockchain.resolveOn`? Would it make sense? You could also take it with `const { resolveOn } = Blockchain`, if needed.

File: tests/bundle-test.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ const {
   BalanceUtils,
 } = kilt
 
-const resolveOn = Blockchain.IS_IN_BLOCK
+kilt.config({ submitTxResolveOn: Blockchain.IS_IN_BLOCK })
```
Comment (ntn-x2) [inline tests/bundle-test.ts:39]: Love it!

**Review by ntn-x2 — APPROVED** [review]
Just a minor point, but otherwise I love it!

**Pull request merged**